pkg/loops/composite_loop.go

```diff
@@ -129,6 +131,10 @@ func (r *CompositeLoop) Parse(data string, args ...string) (int, error) {
 				r.SubLoops = append(r.SubLoops, *newChild)
 			} else {
 				if repeatIdx == 0 && rules.IsMaskRequired(rule.Mask) {
+					length := util.GetRecordSize(data[read:])
+					if length > 0 {
+						log.Println("Parse Error:(" + rule.Name + "), (" + err.Error() + "), (" + data[read:read+int(length)] + ")")
```

Comment:
Library code shouldn't log. Can we return an error instead? 

```suggestion
						fmt.Errorf("%s parsing of %s failed: %v", rule.Name, data[read:read+int(length)], err)
```

Reply:
sorry, make sense

Reply:
fixed again